### PR TITLE
Player Tracking Proto and Go stubs

### PR DESCRIPTION
**Pull request by markmandel**
Implementation of the alpha functions for player tracking in the proto definition.

This includes a interface and non-working stub for the Go gRPC clients as well.

Copyright notices where updated to 2020 on the other languages as they were regenerated during the process.

Work on #1033

### CODE REVIEW

File: pkg/sdkserver/alpha/localsdk.go [hunk 1]
```diff
@@ -14,7 +14,10 @@
 
 package alpha
 
-import "agones.dev/agones/pkg/sdk/alpha"
+import (
+	"agones.dev/agones/pkg/sdk/alpha"
+	"golang.org/x/net/context"
```
Comment (roberthbailey) [inline pkg/sdkserver/alpha/localsdk.go]: Is there a reason you can't just import "context" (https://golang.org/pkg/context/)?

File: pkg/sdkserver/alpha/sdkserver.go [hunk 2]
```diff
@@ -26,3 +29,28 @@ type SDKServer struct{}
 func NewSDKServer() *SDKServer {
 	return &SDKServer{}
 }
+
+// PlayerConnect should be called when a player connects
+func (s SDKServer) PlayerConnect(ctx context.Context, id *alpha.PlayerId) (*alpha.Empty, error) {
+	panic("implement me")
+}
+
+//PlayerDisconnect should be called when a player disconnects
```
Comment (roberthbailey) [inline pkg/sdkserver/alpha/sdkserver.go]: nit: Add a space between // and the start of the comment

Comment (roberthbailey) [reply pkg/sdkserver/alpha/sdkserver.go]: Also a super nit, but comments should be formatted as sentences ending with a period.

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: proto/sdk/alpha/alpha.proto [hunk 3]
```diff
@@ -20,4 +20,56 @@ option go_package = "alpha";
...
+message Empty {
+}
+
+// Store a count variable
+message Count {
+    int64 count = 1;
+}
+
+// The unique identifier for a given player
+message PlayerId {
```
Comment (roberthbailey) [inline proto/sdk/alpha/alpha.proto]: I'm wondering if a single string is sufficient for this. What if you have multiple types of "players"? With a single string field the programmer will end up doing some funny string concatenation / parsing logic to shove it all into one field. If we had the option for some generic metadata (labels?), then you could use that instead to namespace different types of players.

I'm not necessarily saying we should change it right now (so we don't fall into the trap of over-designing), but I'm wondering if there is a way for us to get some feedback from folks that would actually use this. Maybe the best way is to get it in as alpha so folks can start playing around with it?

File: pkg/sdkserver/alpha/localsdk.go (hunk 1 above)
Comment (markmandel) [reply pkg/sdkserver/alpha/localsdk.go]: You know what - my IDE always brought in x/net/context and I never questioned it - I assumed there was a dependency there in gRPC... and I was totally wrong.

Excellent spotting. Fixed!

File: proto/sdk/alpha/alpha.proto (hunk 3 above)
Comment (markmandel) [reply proto/sdk/alpha/alpha.proto]: So this is the design that was outlined [in the accompanying ticket ](#1033), and nobody disagreed with it. (Also, in 1:1 discussions, everyone seemed okay with the design).

_puts opinionated hat on_ :tophat: I don't think it's Agone's job to be storing/transferring data about players - that's why we only use a playerId. Any other metadata about players a game system wants to store, they should store that separately, and attach it in that system to this unique id that is passed between Agones and other systems for identification purposes.

Comment (roberthbailey) [reply proto/sdk/alpha/alpha.proto]: Seems reasonable. It sounds like you've thought about this and gotten some feedback.

**Comment by markmandel** [comment]
Gentle bump on review please

**Comment by markmandel** [comment]
Can I please get review on this, and #1324 , because I'm currently stuck on player tracking until these get merged.

**Comment by markmandel** [comment]
Thanks @devjgm but I need approval from someone with write access - so @pooneh-m or @roberthbailey at this point.

**Comment by roberthbailey** [comment]
Looking now.

File: pkg/sdkserver/alpha/sdkserver.go [hunk 4]
```diff
@@ -14,15 +14,43 @@
 
 package alpha
 
-import "agones.dev/agones/pkg/sdk/alpha"
+import (
+	"agones.dev/agones/pkg/sdk/alpha"
+	"golang.org/x/net/context"
```
Comment (roberthbailey) [inline pkg/sdkserver/alpha/sdkserver.go]: just "context" as in the previous file.

File: pkg/sdkserver/alpha/localsdk.go [hunk 5]
```diff
@@ -14,20 +14,49 @@
...
 	return &LocalSDKServer{}
 }
 
-// Close tears down all the things
+// Close tears down all the things.
 func (l *LocalSDKServer) Close() {
 	// placeholder in case things need to be shut down
 }
+
+// PlayerConnect should be called when a player connects
```
Comment (roberthbailey) [inline pkg/sdkserver/alpha/localsdk.go]: nit: end comment sentence with a period.

(I was going to let this slide until I found other things to comment on. As long as you are fixing those, this should be an easy addition)

File: pkg/sdkserver/alpha/sdkserver.go (hunk 4 above)
Comment (markmandel) [reply pkg/sdkserver/alpha/sdkserver.go]: Thank you!

**Pull request merged**